addons/sale_purchase/models/sale_order.py

```diff
@@ -71,12 +71,14 @@ class SaleOrderLine(models.Model):
     _inherit = 'sale.order.line'
 
     purchase_line_ids = fields.One2many('purchase.order.line', 'sale_line_id', string="Generated Purchase Lines", readonly=True, help="Purchase line generated by this Sales item on order confirmation, or when the quantity was increased.")
-    purchase_line_count = fields.Integer("Number of generated purchase items", compute='_compute_purchase_count', compute_sudo=True)
+    purchase_line_count = fields.Integer("Number of generated purchase items", compute='_compute_purchase_count')
```

Comment:
have you try to add : `groups='purchase.group_purchase_user'` ?

Reply:
Yes and we have an "Error: Unknown field purchase_line_count in domain"
due to stock_dropshipping : 
https://github.com/odoo/odoo/commit/6236ed1a5b138c5165929aa3f5c1d1f4f6aa061c#diff-2f6cca7f38255368e4a54b0a19406aa4